src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Interlocked.cs

```diff
@@ -35,18 +35,21 @@ public static short CompareExchange(ref short location1, short value, short comp
         public static int CompareExchange(ref int location1, int value, int comparand)
         {
 #if TARGET_X86 || TARGET_AMD64 || TARGET_ARM64 || TARGET_RISCV64
-            return CompareExchange(ref location1, value, comparand);
+            return CompareExchange(ref location1, value, comparand); // Must expand intrinsic
 #else
             return RuntimeImports.InterlockedCompareExchange(ref location1, value, comparand);
```

Comment:
We should get the non-intrinsic int8/int16/int32 variants onto the same portable null reference checking plan. (It can be done in a separate PR if you do not want to do it in this one.)

Reply:
There's already PR #97792 that completely removes them on ARM, so I didn't want to create more conflicts than necessary.

Reply:
The portable path is still useful to have for other platforms.

The merge conflicts with the other PR should not be too bad to resolve.

Reply:
(and #99011 moves the int8/int16 fallbacks completely to managed so only int16 will be left on the native side)

Reply:
`int8/int16` are handled by #99011 which is waiting on having me or somebody else fix the JIT assert there (which isn't exactly simple according to @SingleAccretion)

Reply:
Done.